utils/Game.py

```diff
@@ -1,39 +1,41 @@
-from utils.Logger import Logger
+from utils.ScoreLogger import ScoreLogger
 import gym
 
 class Game:
-    def __init__(self, steps, strategy):
-        self.steps = steps
+    def __init__(self, episodes, strategy, render=False):
+        self.episodes = episodes
         self.strategy = strategy
-        self.logger = Logger()
+        self.logger = ScoreLogger()
+        self.render = render
 
-        self.env = gym.make('CartPole-v0')
-        self.env.reset()
+        self.env = gym.make('CartPole-v1')
 
-    def log(self, observation):
-        self.logger.log(
-            self.strategy.get_pole_position(observation),
-            self.strategy.get_pole_velocity(observation),
-            self.strategy.get_pole_angle(observation),
-            self.strategy.get_pole_tip_velocity(observation)
-        )
-
-    def print_logs(self):
-        print(self.logger.get_positions())
-        print(self.logger.get_velocities())
-        print(self.logger.get_angles())
-        print(self.logger.get_tip_velocities())
+    def log(self, score):
+        self.logger.log(score)
 
     def plot(self, title=""):
         self.logger.plot(title)
 
     def play(self):
-        observation, reward, done, info = self.env.step(0)
-        for _ in range(self.steps):
-            self.env.render()
-            action = self.strategy.calculate(observation)
-            observation, reward, done, info = self.env.step(action)
-            self.log(observation)
+        episode = 0
+        while episode < self.episodes:
+            episode += 1
+
+            self.env.reset()
+            observation, reward, done, info = self.env.step(0)
+
+            step = 0
+
+            while True:
+                step += 1
+                if self.render:
+                    self.env.render()
+                action = self.strategy.calculate(observation)
+                observation, reward, done, info = self.env.step(action)
+                if done:
+                    self.log(step)
```

Comment:
I feel like we should keep the original logger to log the observations and also have the score logger

Reply:
Instead of replacing the original logger that is, and I would log the observations for the final step

Reply:
Added the original logger back in